pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -275,7 +275,7 @@ func readInstanceID(searchOrder string) (string, error) {
 		instanceID := string(idBytes)
 		instanceID = strings.TrimSpace(instanceID)
 		glog.V(3).Infof("Got instance id from %s: %s", instanceIDFile, instanceID)
-		if instanceID != "" {
+		if instanceID != "" && instanceID != "iid-datasource-none" {
```

Comment:
some providers seem to return a fake id at least some of the time, so account for that - http://cloudinit.readthedocs.io/en/latest/topics/datasources/fallback.html